src/reporting/clients/ODataClient.ts

```diff
@@ -0,0 +1,86 @@
+    const url = `${this.basePath}/odata/${encodeURIComponent(reportId)}`;
+    const requestOptions: RequestInit = this.createRequest("GET", accessToken);
+    return this.fetchData<ODataResponse>(url, requestOptions);
+  }
+
+  public async getODataReportEntityPage(accessToken: AccessToken, reportId: string, odataItem: ODataItem, sequence: number) : Promise<ODataEntityResponse> {
+    const segments = odataItem.url.split('/');  // region, manifestId, entityType
+    if (segments.length !== 3) {
+      throw new RequiredError(
+        'odataItem',
+        'odata item was invalid when calling updateCalculatedProperty.',
+      );
+    }
+    const url = `${this.basePath}/odata/${encodeURIComponent(reportId)}/${odataItem.url}?sequence=${encodeURIComponent(sequence)}`;
+    const requestOptions: RequestInit = this.createRequest("GET", accessToken);
+    return this.fetchData(url, requestOptions);
+  }
+
+  public async getODataReportEntities(accessToken: AccessToken, reportId: string, odataItem: ODataItem): Promise<Array<ODataEntityValue>> {
+    const segments = odataItem.url.split('/');  // region, manifestId, entityType
+    if (segments.length !== 3) {
+      throw new RequiredError(
+        'odataItem',
+        'odata item was invalid when calling updateCalculatedProperty.',
+      );
+    }
+    const reportData: Array<ODataEntityValue> = [];
+    const oDataReportEntitiesIt = this.getODataReportEntitiesIterator(accessToken, reportId, odataItem);
+    for await(const oDataReportEntity of oDataReportEntitiesIt) {
+      reportData.push(oDataReportEntity);
+    }
+    return reportData;
+  }
+
+  public getODataReportEntitiesIterator(accessToken: AccessToken, reportId: string, odataItem: ODataItem): EntityListIterator<ODataEntityValue> {
+    const url = `${this.basePath}/odata/${encodeURIComponent(reportId)}/${odataItem.url}`;
+    return new EntityListIteratorImpl(async () => getEntityCollectionPage<ODataEntityValue>(
+      url,
+      this.createRequest("GET", accessToken),
+      async (url: string, requestOptions: RequestInit): Promise<Collection<ODataEntityValue>> => {
+        const response: ODataEntityResponse = await this.fetchData<ODataEntityResponse>(url, requestOptions);
+        const link: PagedResponseLinks = {
+          self: {
+            href: ""
+          }
+        };
+        if(response["@odata.nextLink"]) {
+          link.next = {
+            href: response["@odata.nextLink"]
+          }
+        }
+        return {
+          values: response.value,
+          _links: link,
+        }
+    }));
+  }
+
+  public async getODataReportMetadata(accessToken: AccessToken, reportId: string): Promise<Response> {
+    const url = `${this.basePath}/odata/${encodeURIComponent(reportId)}/$metadata`;
+    const requestOptions: RequestInit = this.createRequest("GET", accessToken);
+    return this.fetch(url, requestOptions).then((response) => {
```

Comment:
I think `this.fetchData` already has such error handling, so replace this `fetch` with a call to `this.fetchData`.

Reply:
Replaced with fetchData